### PR TITLE
refactor: remove migration application

### PR DESCRIPTION
**Pull request by Zelldon**
### Which problem does the PR fix?

Removes obsolete migration application. Was already removed here https://github.com/camunda/camunda/pull/24620

### What's in this PR?

Remove the config and init-container

### Checklist

Please make sure to follow our [Contributing Guide](../blob/main/docs/contributing.md).

**Before opening the PR:**

- [x] In the repo's root dir, run `make go.update-golden-only`.
- [x] There is no other open [pull request](../pulls) for the same update/change.
- [ ] Tests for charts are added (if needed).
- [ ] In-repo [documentation](../blob/main/docs/contributing.md#documentation) are updated (if needed).

**After opening the PR:**

- [ ] Did you sign our CLA (Contributor License Agreement)? It will show once you open the PR.
- [ ] Did all checks/tests pass in the PR?

### CODE REVIEW

File: charts/camunda-platform-alpha/test/unit/operate/golden/deployment.golden.yaml [hunk 1]
```diff
@@ -41,33 +41,6 @@ spec:
       imagePullSecrets:
         []
       initContainers:
```
Comment (houssain-barouni) [inline charts/camunda-platform-alpha/test/unit/operate/golden/deployment.golden.yaml:43]:
```suggestion
      initContainers:
        []
```

**Review by houssain-barouni — COMMENTED** [review]
In the failing CI, I still see migration container still referenced
```
  Warning  BackOff      4m59s (x69 over 19m)  kubelet            Back-off restarting failed container migration in pod integration-operate-7b966d5f99-dsxgr_camunda-pr-2573-intg-alpha-run-11859469969-sfx-2db799-upgrade(1462297a-83c5-4f30-aaeb-cafa5bf973fe)
Defaulted container "operate" out of: operate, migration (init)
```

Also operate templates were removed in https://github.com/camunda/camunda-platform-helm/pull/2554 by @aabouzaid

**Comment by Zelldon** [comment]
So to me it looks like this PR is obsolete right? @houssain-barouni @aabouzaid there is no migration app as init container anymore in the core statefulset.

**Comment by aabouzaid** [comment]
@ChrisKujawa that's correct. No `migration` int container for Core so far.

**Pull request closed without merging**